GitExtensions/Program.cs

```diff
@@ -368,5 +371,16 @@ private static bool LocateMissingGit()
 
             return false;
         }
+
+        private static void RegisterServices(ServiceContainer serviceContainer)
+        {
+            serviceContainer.AddService(typeof(IFileSystem), new FileSystem());
+            serviceContainer.AddService(typeof(IGitDirectoryResolver), new GitDirectoryResolver((IFileSystem)serviceContainer.GetService(typeof(IFileSystem))));
+            serviceContainer.AddService(typeof(IRepositoryDescriptionProvider), new RepositoryDescriptionProvider((IGitDirectoryResolver)serviceContainer.GetService(typeof(IGitDirectoryResolver))));
+            serviceContainer.AddService(typeof(IAppTitleGenerator), new AppTitleGenerator((IRepositoryDescriptionProvider)serviceContainer.GetService(typeof(IRepositoryDescriptionProvider))));
+            serviceContainer.AddService(typeof(IWindowsJumpListManager), new WindowsJumpListManager((IRepositoryDescriptionProvider)serviceContainer.GetService(typeof(IRepositoryDescriptionProvider))));
```

Comment:
More lines but seems clearer to me:
```suggestion
            FileSystem fileSystem = new();
            GitDirectoryResolver gitDirectoryResolver = new(fileSystem);
            RepositoryDescriptionProvider repositoryDescriptionProvider = new(gitDirectoryResolver);

            serviceContainer.AddService(typeof(IFileSystem), fileSystem);
            serviceContainer.AddService(typeof(IGitDirectoryResolver), gitDirectoryResolver);
            serviceContainer.AddService(typeof(IRepositoryDescriptionProvider), repositoryDescriptionProvider);
            serviceContainer.AddService(typeof(IAppTitleGenerator), new AppTitleGenerator(repositoryDescriptionProvider));
            serviceContainer.AddService(typeof(IWindowsJumpListManager), new WindowsJumpListManager(repositoryDescriptionProvider));
```

Reply:
❗ Basically, there should be a generic (extension) method which allows to write it shorter:
`serviceContainer.GetService<IFileSystem>()`

❔ Phillippe's suggestion is good and even slightly safer. Is there a _best practice_ which recommends using `GetService` during the registration?

Reply:
I am planning to bring extensions such as `GetService<T>` and `GetRequiredService<T>` in follow up PRs.

Reply:
> Is there a _best practice_ which recommends using `GetService` during the registration?

I'm not aware of any.